### ISSUE TITLE
Integrate Pod::To::Cached

### ISSUE THREAD

**Issue by antoniogamiz** [noisegul/perl6-p6doc#17]
## Problem

I have tried `p6doc` and it takes approximately 3 seconds to actually see the output. I think this time should be highly reduced. Programs similar to this (like `man` command`) gives you the result immediately. 

## Solution

I do not know if you have seen [Pod::To::Cached][1], made by @finanalyst. This module takes a pod collection and creates a cache and when all files have been compiled (only once), it lets you consult the pod structure immediately.

From my experience with [Perl6::Documentable][2], I can say this will reduce the time significantly. With all the pod collection precompiled (almost 400 files), it only takes 5 seconds to process them all.

So, I think let the user use a precompiled cache as an additional option would be helpful.

How could this be done? I think this is related to #14. As you may know, when a pod is processed in [Perl6::Documentable][2], several `Perl6::Documentable` objects are created referencing the different definitions found and that is exactly what the user will want to look up using: `p6doc Str.split`, for instance.

So I can add an additional attribute to `Perl6::Documentable` class, indicating the index associated to one definition in the pod structure. With this attribute and the index, the search time will be virtually constant and the results can be output immediately.

The index would be obtained from [Perl6::Documentable][2] as is said in #14. It only needs an entry for each and every file and associated to that entry, a list with the names and the extension of each and every definition found.

This is only an idea that have crossed my mind, please let me know what you think.

[1]: https://github.com/finanalyst/pod-cached
[2]: https://github.com/antoniogamiz/Perl6-Documentable

**Comment by noisegul** [comment]
Hey, thanks for the input and for lining that out!
As you know I'm currently taking the stab at integrating Perl6::Documentable, what that means is that nothing from way it currently works would stay.

However as you have noticed there are some problems with integration and open questions.
A big factor is that this apparently means that Windows support would be dropped, as it appears that it's `Pod::To::Cached` that causes the failure to install on Windows, and there is the status of usage together with Rakudo-Star in general. For reference: https://github.com/antoniogamiz/Perl6-Documentable/issues/48

I should open another issue for that discussion however, after that has been investigated.

**Comment by noisegul** [comment]
I'll close this issue for now and then create more specific ones related to the approach once I reach the phase of optimizations (after `0.1.0` release), should this one be pursued.
